### PR TITLE
interesting: plumb external informers to aggregator

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

File: pkg/cmd/server/origin/aggregator.go [hunk 1]
```diff
@@ -72,14 +75,18 @@ func (c *MasterConfig) createAggregatorConfig(genericConfig genericapiserver.Rec
 	}
 
 	aggregatorConfig := &aggregatorapiserver.Config{
-		GenericConfig: &genericConfig,
+		GenericConfig: &genericapiserver.RecommendedConfig{
+			Config:                genericConfig,
+			SharedInformerFactory: externalInformers,
```
Comment (sttts) [inline pkg/cmd/server/origin/aggregator.go:80]: why not instatiate outside of createAggregatorConfig ?

Comment (sttts) [reply pkg/cmd/server/origin/aggregator.go:80]: ~~only a nit~~

**Comment by soltysh** [comment]
Manually cherry-picked.

File: pkg/cmd/server/origin/master.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func (c *MasterConfig) newOAuthServerHandler(genericConfig *apiserver.Config) (h
 }
 
 func (c *MasterConfig) withAggregator(delegateAPIServer apiserver.DelegationTarget, kubeAPIServerConfig apiserver.RecommendedConfig, apiExtensionsInformers apiextensionsinformers.SharedInformerFactory) (*aggregatorapiserver.APIAggregator, error) {
-	aggregatorConfig, err := c.createAggregatorConfig(kubeAPIServerConfig)
+	aggregatorConfig, err := c.createAggregatorConfig(kubeAPIServerConfig.Config, c.ClientGoKubeInformers)
```
Comment (sttts) [inline pkg/cmd/server/origin/master.go:197]: Why isn't `kubeAPIServerConfig.SharedInformerFactory` set with a non-nil value?

Comment (sttts) [reply pkg/cmd/server/origin/master.go:197]: This change looks suspicious, now hiding another error.

File: pkg/cmd/server/origin/aggregator.go (hunk 1 above)
Comment (mfojtik) [reply pkg/cmd/server/origin/aggregator.go:80]: @sttts this is what kube does, wanted to copy their pattern.

File: pkg/cmd/server/origin/master.go (hunk 2 above)
Comment (sttts) [reply pkg/cmd/server/origin/master.go:197]: The pattern is good. But we have the same problem with `kubeAPIServerConfig`. It should have the informer field set.

**Pull request closed without merging**